src/validation/manifest.js

```diff
@@ -241,23 +241,61 @@ const validateValue = (value, ctx) => {
   }
 }
 
-const validateDataSourceForNetwork = (dataSources, protocol) =>
+// Transforms list of data sources like this:
+// [
+//   { name: 'contract0', kind: 'ethereum', network: 'mainnet' },
+//   { name: 'contract1', kind: 'ethereum', network: 'mainnet' },
+//   { name: 'contract2', kind: 'ethereum', network: 'xdai' },
+//   { name: 'contract3', kind: 'near', network: 'near-mainnet' },
+// ]
+//
+// Into Immutable JS structure like this:
+// {
+//   ethereum: {
+//     mainnet: ['contract0', 'contract1'],
+//     xdai: ['contract2'],
+//   },
+//   near: {
+//     'near-mainnet': ['contract3'],
+//   },
+// }
+const dataSourceListToMap = dataSources =>
   dataSources
-    .filter(dataSource => protocol.isValidKindName(dataSource.kind))
     .reduce(
-      (networks, dataSource) =>
-        networks.update(dataSource.network, dataSources =>
-          (dataSources || immutable.OrderedSet()).add(dataSource.name),
+      (protocolKinds, dataSource) =>
+        protocolKinds.update(dataSource.kind, networks =>
+          (networks || immutable.OrderedMap()).update(dataSource.network, dataSourceNames =>
+            (dataSourceNames || immutable.OrderedSet()).add(dataSource.name)),
         ),
       immutable.OrderedMap(),
     )
 
-const validateDataSourceNetworks = (value, protocol) => {
+const validateDataSourceProtocolAndNetworks = (value, protocol) => {
```

Comment:
Do you need the `protocol` argument here, it seems unused?

Reply:
You're correct! Just removed it 🔫